### ISSUE TITLE
problem with - import {ROUTE_PROVIDERS, Router, etc.}

### ISSUE THREAD

**Issue by mLaird** [TypeStrong/atom-typescript#712]
The import in the title generates - Error Module '"angular2/router"' has no exported member 'ROUTER_PROVIDERS'  Compile fails, but emit occurs, and the page loads successfully.  When I remove ROUTE_PROVIDERS,  compile and emit occur, but the page does not load.  I'm using angular2 alpha.45.  Bottom line:  ROUTE_PROVIDERS is now necessary, but some part of atom or atom-typescript does not recognize it.

**Comment by basarat** [comment]
> Bottom line: ROUTE_PROVIDERS is now necessary, but some part of atom or atom-typescript does not recognize it.

Sounds like an issue in angular's TypeScript definition.

**Comment by basarat** [comment]
Please share project :rose:

**Comment by mLaird** [comment]
Thank you.  You can view the relevant code here:

http://stackoverflow.com/questions/33681486/angular2-routing-alpha-46-routes-forward-with-a-console-error-but-not-back-to

The problem statement has 2 issues, one of which is the atom-typescript
error message.

On Wed, Nov 11, 2015 at 12:36 AM, Basarat Ali Syed <notifications@github.com
> wrote:

> Please share project [image: :rose:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TypeStrong/atom-typescript/issues/712#issuecomment-155672173>
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by basarat** [comment]
> Thank you.  You can view the relevant code here:

more along the lines of `zip download` or `git checkout` please. I do not have an angular setup.

**Comment by mLaird** [comment]
I don't know how to create a zip download or do a git checkout.
I have attached relevant files.

On Thu, Nov 12, 2015 at 5:59 PM, Basarat Ali Syed <notifications@github.com>
wrote:

> Thank you. You can view the relevant code here:
>
> more along the lines of zip download or git checkout please. I do not
> have an angular setup.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TypeStrong/atom-typescript/issues/712#issuecomment-156263876>
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by basarat** [comment]
@mLaird sorry mike, the files aren't attached to your comment (if you reply by email the attachments are lost ... and attaching zip to github comments isn't allowed anyways). By git checkout I meant create a github repository :rose: Alternatively share over dropbox

**Comment by mLaird** [comment]
I'm trying to share the project via Google Drive.  The project has been
created in Windows 10, so the directories have a lot of Windows overhead.
The individual files are a few kb, but the whole thing is 13.1 Mb.  Hope
this works for you.  You should be able to access and download the project
at this link:

https://drive.google.com/folderview?id=0BxDxP4eSMEPnYktsX3JFeVdJODA&usp=sharing

I have added 2 more Angular2 arrays of directives, so atom-typescript now
has 3 error messages of 'cannot find',  but the page continues to load just
fine.

Hope this helps you.

On Thu, Nov 12, 2015 at 9:22 PM, Basarat Ali Syed <notifications@github.com>
wrote:

> @mLaird <https://github.com/mLaird> sorry mike, the files aren't attached
> to your comment (if you reply by email the attachments are lost ... and
> attaching zip to github comments isn't allowed anyways). By git checkout I
> meant create a github repository [image: :rose:] Alternatively share over
> dropbox
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TypeStrong/atom-typescript/issues/712#issuecomment-156300582>
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by basarat** [comment]
Two issues. 
* The version of angular you have there is too old. alpha 36 . It does not ship *with* definitions. 
![image](https://cloud.githubusercontent.com/assets/874898/11172005/00b455ee-8c52-11e5-8323-0ebc68f05a64.png)

* The tsconfig was wrong. `filesGlob` was miss-spelled and you really should exclude `node_modules`, this made the performance horrible. Check faq https://github.com/TypeStrong/atom-typescript/blob/master/docs/faq.md#for-really-large-projects-atom-typescript-gets-slow :rose: . Fixed: 

```json
{
    "version": "1.5.0",
    "compilerOptions": {
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "target": "es5",
        "module": "commonjs",
        "declaration": false,
        "noImplicitAny": false,
        "removeComments": true,
        "noLib": false,
        "outDir": "dist/js/cjs"
    },
    "filesGlob": [
        "./src/app.ts"
    ],
    "exclude": [
        "node_modules"
    ],
    "files": [
        "./src/app.ts"
    ]
}
```

Let us know if its still unclear :rose:

**Comment by mLaird** [comment]
Thank you, I am still confused, and still have 3 atom-typescript error
messages about 'no exported member'

I found 3 tsconfig.json files in my project.  I presume you mean the one in
the angular2-oPost project file.  I added the new items
from your email above.  Except I did not add the "version": "1.5.0" line.
I don't understand that line. I'm using atom-typescript 7.9.0 at
the moment, and as you know, it changes frequently. The other 2
tsconfig.json files are buried down inside the .atom directory.  I presume
atom-typescript uses these 2 files, so I did not change them.

Regarding the angular2 version,  the index.html that I sent you in my
project directory calls for alpha.46,  as follows:

<script src="https://code.angularjs.org/2.0.0-alpha.46/angular2.dev.js"></script>

<script src="https://code.angularjs.org/2.0.0-alpha.46/router.dev.js"></script>


I looked for angular2 files in my project and did not find any.  What led
you to believe/find the project uses alpha36, where is that file,
and why has it not changed to alpha.46 per the script in index.html?

I get no error message when I hover over 'angular2/router', like in the
example you sent above.  When I hover it says 'module'.



On Sun, Nov 15, 2015 at 7:08 PM, Basarat Ali Syed <notifications@github.com>
wrote:

> Closed #712 <https://github.com/TypeStrong/atom-typescript/issues/712>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/TypeStrong/atom-typescript/issues/712#event-464931702>
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy
